schema_editor/app/scripts/schemas/schemas-service.js

```diff
@@ -64,7 +96,7 @@
                             },
                             enumSource: [{
                                     source: 'target',
-                                    title: fieldData.referenceTarget + ' {{i}}',
+                                    title: displayProperties.join(' '),
```

Comment:
Oh, gotcha. This seems fine for now. We could probably even get by with just limiting the total length of the dropdown items using css.